Sources/FaceLiveness/FaceDetection/BlazeFace/DetectedFace.swift

```diff
@@ -14,53 +14,78 @@ struct DetectedFace {
     let rightEye: CGPoint
     let nose: CGPoint
     let mouth: CGPoint
+    let rightEar: CGPoint
+    let leftEar: CGPoint
 
     let confidence: Float
 
-    func boundingBoxFromLandmarks() -> CGRect {
-        let eyeCenterX = (leftEye.x + rightEye.x) / 2
-        let eyeCenterY = (leftEye.y + rightEye.y) / 2
-
-        let cx = (nose.x + eyeCenterX) / 2
-        let cy = (nose.y + eyeCenterY) / 2
-
-        let ow = sqrt(pow((leftEye.x - rightEye.x), 2) + pow((leftEye.y - rightEye.y), 2)) * 2
-        let oh = 1.618 * ow
-        let minX = cx - ow / 2
-        let minY = cy - oh / 2
-
-        let rect = CGRect(x: minX, y: minY, width: ow, height: oh)
+    func boundingBoxFromLandmarks(ovalRect: CGRect) -> CGRect {
+        let alpha = 2.0
+        let gamma = 1.8
+        let ow = (alpha * pupilDistance + gamma * faceHeight) / 2
+        var cx = (eyeCenterX + nose.x) / 2
+        
+        if ovalRect != CGRect.zero {
+            let ovalTop = ovalRect.minY
+            let ovalHeight = ovalRect.maxY - ovalRect.minY
+            if eyeCenterY > (ovalTop + ovalHeight) / 2 {
+                cx = eyeCenterX
+            }
+        }
+        
+        let faceWidth = ow
+        let faceHeight = 1.68 * faceWidth
+        let faceBoxBottom = boundingBox.maxY
+        let faceBoxTop = faceBoxBottom - faceHeight
+        let faceBoxLeft = min(cx - ow / 2, rightEar.x)
+        let faceBoxRight = min(cx + ow / 2, leftEar.x)
```

Comment:
talked to Thadd yesterday about this -- this second line is supposed to be `max()` to match the document

Reply:
good catch, I've updated to `Math.max(...)`